src/main/java/fr/insee/publicenemy/api/infrastructure/ddi/DdiPoguesServiceImpl.java

```diff
@@ -100,6 +104,32 @@ private JsonNode getJsonPogues(@NonNull String questionnaireId) {
                 .bodyToMono(JsonNode.class)
                 .blockOptional().orElseThrow(() -> new PoguesJsonNotFoundException(questionnaireId));
     }
+
+    @Override
+    public List<VariableType> getQuestionnaireVariables(@NonNull String questionnaireId) {
+        String variablesString = webClient.get().uri(poguesUrl + "/api/persistence/questionnaire/{id}/variables", questionnaireId)
+                .retrieve()
+                .onStatus(
+                        HttpStatus.NOT_FOUND::equals,
+                        response -> response.bodyToMono(String.class)
+                                .flatMap(errorMessage -> Mono.error(new PoguesJsonNotFoundException(questionnaireId)))
+                )
+                .onStatus(
+                        HttpStatusCode::isError,
+                        response -> response.bodyToMono(String.class)
+                                .flatMap(errorMessage -> Mono.error(new ServiceException(response.statusCode().value(), errorMessage)))
+                )
+                .bodyToMono(String.class)
+                .blockOptional().orElseThrow(() -> new PoguesJsonNotFoundException(questionnaireId));
+
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            return mapper.readValue(variablesString, new TypeReference<List<VariableType>>(){});
```

Comment:
(IntelliJ warning) Explicit type argument List<VariableType> can be replaced with <>